### ISSUE TITLE
Update request: python3Packages.web3 6.5.0 → 7.6.0

### ISSUE THREAD

**Issue by siraben** [NixOS/nixpkgs#360674]
## Package Information

- Package name: web3
- Latest released version: 7.6.0
- Current version on the unstable channel: 6.5.0
- Current version on the stable/release channel: 6.5.0

PRs to update:
- eth-utils #360668 4.0.0 → 5.1.0
- eth-typing #360667 4.0.0 → 5.0.1

## Checklist

- [ ] Merge/build upon #342749

## Notify maintainers

@hellwolf and @natsukium

**Comment by hellwolf** [comment]
Yea, this is getting ridiculous. I will act on it, either to chase some PR to merge, or take over. But please give me some time.

**Comment by siraben** [comment]
@hellwolf let me know what your plan is going forward, would love to see web3 get updated!

**Comment by hellwolf** [comment]
@siraben Hi! Thanks for pushing for these. I have yet come to this topic. My interest is to keep these packages built on unstable:

- echidna (which has a run-time depency on slither)
- slither (which depends on web3)

If you have a PR for web3 update ready, I can help to review and push community to merge. Otherwise, I will need a few more days to come to this.

**Comment by siraben** [comment]
@hellwolf Ok, great. No rush at all. I've been using web3 from conda at the moment. Ping me if you need any reviews and I can help test it out.

**Comment by hellwolf** [comment]
I started a new PR https://github.com/NixOS/nixpkgs/pull/373563 I will push it forward

**Comment by siraben** [comment]
@hellwolf thanks for your contributions!
